server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -168,7 +168,7 @@ public long getGenId() {
      * Returns an unmodifiable collection of the snapshot ids.
      */
     public Collection<SnapshotId> getSnapshotIds() {
-        return Collections.unmodifiableCollection(snapshotIds.values());
+        return snapshotIds.values();
```

Comment:
I'm not sure we can guarantee that the collection returned by this method is truly unmodifiable if we remove this?

Reply:
I think this is fine because if you look at the constructor of `RepositoryData` we always make sure that `snapshotIds` is unmodifiable which entails that it's value collection is as well :)

Reply:
Right - I did not look at all the ctors, thanks